gmail/gmail.py

```diff
@@ -38,7 +39,15 @@ def connect(self, raise_errors=True):
         #         raise Exception('Connection failure.')
         #     self.imap = None
 
-        self.imap = imaplib.IMAP4_SSL(self.GMAIL_IMAP_HOST, self.GMAIL_IMAP_PORT)
+        # Gmail doesn't allow us to connect to it via IPv6, so the below resolves `imap.gmail.com`
+        # to its IPv4 address and connects using that.
+        # See https://blog.zensoftware.co.uk/2016/05/13/gmail-blocking-mdaemon-email-arriving-via-ipv6/
+        ipv4_gmail_connection = socket.socket(socket.AF_INET)
+        ipv4_gmail_connection.connect((self.GMAIL_IMAP_HOST, self.GMAIL_IMAP_PORT))
```

Comment:
I think @bobbyrenwick means the connection that we make just to resolve the hostname to an IPv4 address - done so in https://github.com/EDITD/gmail/pull/1/commits/d840d1ef5f3aa5e9ada23c132f1fb3a510c1ad50